### PR TITLE
app-misc/nnn: add 4.8

### PR DESCRIPTION
**Pull request by s-hamann**
This PR adds app-misc/nnn-4.8, the latest upstream release.

Due to upstream changes (jarun/nnn#1584), the USE flag `colemak` was renamed to `colemak-dh` and a new USE flag `colemak` was introduced. This change is somewhat confusing, since the behaviour of the existing flag changes, but it follows upstream and seems like the cleanest solution to me.
`colemak` and `colemak-dh` are mutually exclusive USE flags, since both patch the key bindings of nnn in (slightly) different ways.

I also dropped old versions.

### CODE REVIEW

**Comment by Flowdalic** [comment]
Please add the information you wrote in this PR also im the according commit message.

**Comment by s-hamann** [comment]
> Please add the information you wrote in this PR also im the according commit message.

Done.

File: app-misc/nnn/nnn-4.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+inherit bash-completion-r1 flag-o-matic toolchain-funcs xdg
+
+DESCRIPTION="The missing terminal file browser for X"
+HOMEPAGE="https://github.com/jarun/nnn"
+SRC_URI="https://github.com/jarun/nnn/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE="+readline pcre 8contexts icons nerdfonts emoji qsort colemak colemak-dh gitstatus namefirst restorepreview"
```
Comment (juippis) [inline app-misc/nnn/nnn-4.8.ebuild:15]: Let's keep this alphabetically sorted.

File: app-misc/nnn/nnn-4.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+DEPEND="sys-libs/ncurses:=
+	pcre? ( dev-libs/libpcre )
+	readline? ( sys-libs/readline:= )
+	elibc_musl? ( sys-libs/fts-standalone )"
+BDEPEND="virtual/pkgconfig"
+RDEPEND="${DEPEND}"
+REQUIRED_USE="?? ( icons nerdfonts emoji ) ?? ( colemak colemak-dh )"
```
Comment (juippis) [inline app-misc/nnn/nnn-4.8.ebuild:23]: Multi-line this for readability and consistency among the tree.

File: app-misc/nnn/nnn-4.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+	default
+	tc-export CC
+	use elibc_musl && append-flags "-lfts"
+	# When using nnn's bundled patches, the 'install' target should not depend
+	# on 'all'. See: https://github.com/jarun/nnn/issues/1493
+	sed -i -e 's/install: all/install:/' Makefile || die "sed failed"
+	# The Makefile uses O_COLEMAK-DH to control the Colemak-DH patch, but that
+	# does not work with the array approach to make options below. Hence, we
+	# simply rename it to O_COLEMAK_DH.
+	sed -i -e 's/O_COLEMAK-DH/O_COLEMAK_DH/' Makefile || die "sed failed"
```
Comment (juippis) [inline app-misc/nnn/nnn-4.8.ebuild:35]: These can be combined to a single sed call since they edit the same file.

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**